src/CommunityToolkit.Maui/Behaviors/AnimationBehavior.shared.cs

```diff
@@ -0,0 +1,118 @@
+	/// </summary>
+	public static readonly BindableProperty AnimationTypeProperty =
+		BindableProperty.Create(nameof(AnimationType), typeof(BaseAnimation), typeof(AnimationBehavior));
+
+	/// <summary>
+	/// Backing BindableProperty for the <see cref="AnimateCommand"/> property.
+	/// </summary>
+	internal static readonly BindablePropertyKey animateCommandPropertyKey =
+ 			BindableProperty.CreateReadOnly(
+ 				nameof(AnimateCommand),
+ 				typeof(ICommand),
+ 				typeof(AnimationBehavior),
+ 				default,
+ 				BindingMode.OneWayToSource,
+ 				defaultValueCreator: CreateAnimateCommand);
+
+	/// <summary>
+	/// Backing BindableProperty for the <see cref="AnimateCommand"/> property.
+	/// </summary>
+	public static readonly BindableProperty AnimateCommandProperty = animateCommandPropertyKey.BindableProperty;
+
+	TapGestureRecognizer? tapGestureRecognizer;
+
+	/// <summary>
+	/// Command on which to perform the animation.
+	/// </summary>
+	public ICommand AnimateCommand => (ICommand)GetValue(AnimateCommandProperty);
+
+	/// <summary>
+	/// The type of animation to perform
+	/// </summary>
+	public BaseAnimation? AnimationType
+	{
+		get => (BaseAnimation?)GetValue(AnimationTypeProperty);
+		set => SetValue(AnimationTypeProperty, value);
+	}
+
+	/// <inheritdoc/>
+	[MemberNotNull(nameof(tapGestureRecognizer))]
+	protected override void OnAttachedTo(VisualElement bindable)
+	{
+		base.OnAttachedTo(bindable);
+
+		if (!string.IsNullOrWhiteSpace(EventName))
+		{
+			throw new InvalidOperationException($"{nameof(EventName)} must be null. It is not used by {nameof(AnimationBehavior)}");
+		}
+
+		if (bindable is not IGestureRecognizers gestureRecognizers)
+		{
+			throw new InvalidOperationException($"VisualElement does not implement {nameof(IGestureRecognizers)}");
+		}
+
+		if (bindable is not ITextInput)
+		{
+			tapGestureRecognizer = new TapGestureRecognizer();
+			tapGestureRecognizer.Tapped += OnTriggerHandled;
+
+			foreach (var recognizer in gestureRecognizers.GestureRecognizers.OfType<TapGestureRecognizer>())
+			{
+				gestureRecognizers.GestureRecognizers.Remove(recognizer);
+			}
+
+			gestureRecognizers.GestureRecognizers.Add(tapGestureRecognizer);
+		}
+		else
+		{
+			throw new InvalidOperationException($"Animation Behavior can not be attached to {nameof(ITextInput)}");
+		}
+	}
+
+	/// <inheritdoc/>
+	protected override void OnDetachingFrom(VisualElement bindable)
+	{
+		if (tapGestureRecognizer != null)
+		{
+			tapGestureRecognizer.Tapped -= OnTriggerHandled;
+			tapGestureRecognizer = null;
+		}
+
+		base.OnDetachingFrom(bindable);
+	}
+
+	/// <inheritdoc/>
+	protected override async void OnTriggerHandled(object? sender = null, object? eventArgs = null)
+	{
+		await OnAnimate();
+
+		base.OnTriggerHandled(sender, eventArgs);
+	}
+
+	static object CreateAnimateCommand(BindableObject bindable)
+		=> new Command(async () => await ((AnimationBehavior)bindable).OnAnimate().ConfigureAwait(false));
```

Comment:
It would be really nice to make use of `AsyncCommand` here and also when registering the `TapGestureRecognizer`. Do we need to consider having an internal implementation? Or is there a better solution?